src/validators/dataclass.rs

```diff
@@ -335,13 +359,22 @@ impl Validator for DataclassArgsValidator {
                 Err(err) => Err(err),
             }
         } else {
-            Err(ValError::new_with_loc(
-                ErrorType::NoSuchAttribute {
-                    attribute: field_name.to_string(),
-                },
-                field_value,
-                field_name.to_string(),
-            ))
+            // Handle extra (unknown) field
+            // We partially use the extra_behavior for initialization / validation
+            // to determine how to handle assignment
+            match self.extra_behavior {
+                // For dataclasses we allow assigning unknown fields
+                // to match stdlib dataclass behavior
+                ExtraBehavior::Ignore | ExtraBehavior::Unset | ExtraBehavior::Allow => ok(field_value.to_object(py)),
```

Comment:
I think `Ignore` should cause an error here, like it does with models.

Reply:
Changed in [a7179dc](https://github.com/pydantic/pydantic-core/pull/505/commits/a7179dcf18fe6eac9e24c138c7d09a38a70d4fc1)